tests/uitools/qml_quick/functional_tests/FloorFilterFuncTest/tst_floorfilter.qml

```diff
@@ -0,0 +1,474 @@
+                itemId: "49520a67773842f1858602735ef538b5"
+                portal: Portal {
+                    Component.onCompleted: load()
+                    onErrorChanged: {
+                        if (loadStatus === Enums.LoadStatusFailedToLoad) {
+                            console.log(error.message);
+                        }
+                    }
+                    credential: viewerCredentialIndoors
+                    url: "https://indoors.maps.arcgis.com/"
+                }
+            }
+        }
+    }
+
+    Component {
+        id: ffComponent
+        FloorFilter {
+
+        }
+    }
+
+    Component {
+        id: ffMultiComponent
+        FloorFilter {
+            geoView: viewMulti
+            controller: FloorFilterController {
+                id: controllerMulti
+                property SignalSpy spy : SignalSpy {
+                    target: controllerMulti
+                    signalName: "loaded"
+                }
+            }
+        }
+    }
+
+    TestCase {
+        id: floorFilterFuncTest
+        name: "FloorFilterFuncTest"
+        when: windowShown
+        Credential {
+            id: viewerCredentialIndoors
+            // need to add CREDENTIAL_USERNAME and CREDENTIAL_PASSWORD to the Environment variables.
+            username: credential_username
+            password: credential_password
+        }
+
+        function test_collapserDefault() {
+            var ff = createTemporaryObject(ffComponent, foo);
+            var collapser = findChild(ff, "collapser");
+            var zoom = findChild(ff, "zoom");
+            compare(collapser.text, qsTr("Collapse"));
+            compare(collapser.display, AbstractButton.IconOnly);
+            compare(zoom.text, qsTr("Zoom to"));
+            compare(zoom.display, AbstractButton.IconOnly)
+            // wait to give a bit of time on what is happening in the UI
+            wait(1000);
+            mouseClick(collapser);
+            compare(collapser.display, AbstractButton.TextBesideIcon);
+            compare(zoom.display, AbstractButton.TextBesideIcon);
+            wait(1000);
+        }
+
+        // test ff components that change state based on ff realative position to its parent.
+        // x axis changes the gridOuterWrapper layoutDirection switching toolbuttons position with the site/facility component. Collapser icon used (point left or right).
+        function test_visual_changePosition() {
+            var ff = createTemporaryObject(ffComponent, foo);
+            var buildingMenuButton = findChild(ff, "buildingMenuButton");
+            var gridOuterWrapper = findChild(ff, "gridOuterWrapper");
+            var collapser = findChild(ff, "collapser");
+            mouseClick(buildingMenuButton);
+            // set to top left
+            ff.x = 0;
+            ff.y = 0;
+
+            // move ff until its center point is < parent center
+            while(ff.x + ff.width / 2 < ff.parent.width / 2){
```

Comment:
could these while and wait blocks be replaced with try compares?

Reply:
I am not sure how to write them down as `tryCompare()`.
What I am doing here, is moving the floorfilter.x of 20 and then check some of its properties. 
May be a tryVertify()?
```
tryVerify(function() {
                compare(Qt.LeftToRight,  gridOuterWrapper.layoutDirection)
                compare("chevrons-right.svg", Utils.urlFileName(collapser.icon.source.toString()))
                ff.x += 20;
                wait(50);
                return ff.x + ff.width / 2 < ff.parent.width / 2;
});
```

Reply:
i think you probably can't do the inner `compare` lines within a try verify (?) - but maybe that doesn't need to get checked each time around the loop? you could also just return false if those comparisons fail.

Reply:
Tested it out: yes you can use a compare inside a `tryVerify()`, cool!
I can do this:
```
tryVerify(function() {
                if(ff.x + ff.width / 2 >= ff.parent.width / 2)
                    return true;
                compare(Qt.LeftToRight,  gridOuterWrapper.layoutDirection)
                compare("chevrons-right.svg", Utils.urlFileName(collapser.icon.source.toString()))
                ff.x += 20;
                wait(50);
                return false;
            });
```
First `if` is to early return so the `compare` will not fail when the current point is over the center width.
I need to check `compare` at every iterations so I am sure that the `floorfilter` is not behaving strangely in any screen position